### PR TITLE
Added getedid script

### PR DESCRIPTION
**Pull request by DaVukovic**
This is a newer version for the getedid script which contains the changes @MilhouseVH mentioned in the previous PR

### CODE REVIEW

File: packages/sysutils/busybox/scripts/getedid [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  sleep 15
+  reboot
+}
+
+
+# check syslinux.cfg and/or extlinux.conf
+check_file() {
+  # check boot system
+  boot="$([ -d /sys/firmware/efi ] && echo UEFI || echo BIOS)"
+  if [ "$boot" = "BIOS" ]; then
```
Comment (awiouy) [inline packages/sysutils/busybox/scripts/getedid]: would be clearer:
if -d... then 
  boot=... sys_path=...
else
  boot=... sys_path=...
fi

File: packages/sysutils/busybox/scripts/getedid [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    sys_path="/flash"
+  elif [ "$boot" = "UEFI" ]; then
+    sys_path="/flash/EFI/BOOT"
+  fi
+
+  #check which file is available
+  if [ -f "$sys_path"/syslinux.cfg ] && [ -f "$sys_path"/extlinux.conf ];then
+    echo "Your system contains both. A /flash/syslinux.cfg and a /flash/extlinux.conf file"
```
Comment (awiouy) [inline packages/sysutils/busybox/scripts/getedid]: both a

File: packages/sysutils/busybox/scripts/getedid [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    echo "Dual boot systems aren't supported"
+    exit 2
+  fi
+}
+
+
+check_content() {
+  # check if changes are already made to $file and exit if yes
+  if [ "$(grep "APPEND" $file)" = " APPEND boot=LABEL=System disk=LABEL=Storage ssh quiet" ]; then
+    echo "You alread did some changes to $file. Exiting."
```
Comment (awiouy) [inline packages/sysutils/busybox/scripts/getedid]: already

Comment (awiouy) [inline packages/sysutils/busybox/scripts/getedid]: i do not understand, how can you know that the user made exactly these changes?

File: packages/sysutils/busybox/scripts/getedid [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+  if [ "$gpu" = "intel" ]; then
+    check_file
+    mount_rw
+    if [ -f "$file".old ]; then
+      cp "$file".old "$file"  # cp because of keeping the backup...just in case
+    else
+      echo "You didn't created a custom EDID yet or you didn't use this script for this task."
+      echo "Therefore we can't ensure the script working properly."
+      echo "Exiting"
+      exit 3
```
Comment (awiouy) [inline packages/sysutils/busybox/scripts/getedid]: you exit in rw mode, move mount_rw between if and else

File: packages/sysutils/busybox/scripts/getedid [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+    fi
+  elif [ "$gpu" = "nvidia" ]; then
+    if [ -f /storage/.config/xorg.conf ]; then
+      mv /storage/.config/xorg.conf /storage/.config/xorg.le.backup
+    else
+      echo "You don't have a xorg.conf in your .config folder. Exiting"
+      exit 7
```
Comment (awiouy) [inline packages/sysutils/busybox/scripts/getedid]: here, too: you exit in rw mode

File: packages/sysutils/busybox/scripts/getedid [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+
+  # get port
+  nv_port="$(cat /var/log/Xorg.0.log | grep -i -w connected | grep -i -o "dfp-[0-9]" | head -n 1)"
+  nvidia-xconfig --extract-edids-from-file=/var/log/Xorg.0.log --extract-edids-output-file=/storage/.config/edid.bin 1>/dev/null
+
+  # set mode debug back to false
+  sed -i 's/"ModeDebug" "true"/"ModeDebug" "false"/g' /storage/.config/xorg.conf
+  
+  # set port and uncomment lines
+  sed -i "s/#    Option         \"ConnectedMonitor\" \"DFP-0\"/    Option         \"ConnectedMonitor\" \"$nv_port\"/g" /storage/.config/xorg.conf
```
Comment (awiouy) [inline packages/sysutils/busybox/scripts/getedid]: Are the seded lines invariable (they exist and are never edited)?

File: packages/sysutils/busybox/scripts/getedid (hunk 3 above)
Comment (DaVukovic) [inline packages/sysutils/busybox/scripts/getedid]: AFAIK this is the standard content of the APPEND line after a fresh install

File: packages/sysutils/busybox/scripts/getedid (hunk 6 above)
Comment (DaVukovic) [inline packages/sysutils/busybox/scripts/getedid]: The lines I change by using sed  can ofc be edited by the user before. But therefore I check if a xorg.conf already exists and exit if yes. If the user already did some changes to the xorg.conf, he's also able to do that a 2nd time ;) and then won't need that script

File: packages/sysutils/busybox/scripts/getedid [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+  # check boot system
+  boot="$([ -d /sys/firmware/efi ] && echo UEFI || echo BIOS)"
+  if [ "$boot" = "BIOS" ]; then
+    sys_path="/flash"
+  elif [ "$boot" = "UEFI" ]; then
+    sys_path="/flash/EFI/BOOT"
+  fi
+
+  #check which file is available
+  if [ -f "$sys_path"/syslinux.cfg ] && [ -f "$sys_path"/extlinux.conf ];then
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid]: `"$sys_path"/syslinux.cfg` ->`"$sys_path/syslinux.cfg"`
`"$sys_path"/extlinux.conf` ->`"$sys_path/extlinux.conf"`

and the same wherever this has been repeated. Presumably it works as it is, but I find it a little odd.

Comment (DaVukovic) [inline packages/sysutils/busybox/scripts/getedid]: ok, will do

File: packages/sysutils/busybox/scripts/getedid (hunk 3 above)
Comment (awiouy) [inline packages/sysutils/busybox/scripts/getedid]: maybe != instead of = then, no?

Comment (DaVukovic) [inline packages/sysutils/busybox/scripts/getedid]: True ;)

**Comment by DaVukovic** [comment]
Did some changes ^^

**Comment by MilhouseVH** [comment]
Don't forget to install the script... https://github.com/LibreELEC/LibreELEC.tv/blob/master/packages/sysutils/busybox/package.mk#L158-L165

**Comment by MilhouseVH** [comment]
You should also `chmod 775` the `getedid` file, like the other scripts in the same folder.

File: packages/sysutils/busybox/scripts/getedid [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+# exit 7 = no xorg.conf in /storage/.config
+
+
+# Help message and usage explanation
+help() {
+echo "This script generates a custom EDID depending on your GPU"
+echo ""
+echo "To check which GPU you are using, use: ./getedid gpu"
+echo ""
+echo "To create a custon EDID, just use this script like: ./getedid start"
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid]: `getedid start` is a little odd, what am I starting? Why not `getedid create` as this more in keeping with the other `delete` and `restore` options.

File: packages/sysutils/busybox/scripts/getedid [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+  fi
+  sys_reboot
+}
+
+
+intel() {
+  #check which output is connnected:
+  for i in /sys/class/drm/*; do
+    if [ "$(cat "$i"/status 2>/dev/null)" = "connected" ]; then
+      hdmi="$(echo "$i" | cut -d / -f 5 | sed 's/card[0-9]-//g')"
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid]: `break`?

File: packages/sysutils/busybox/scripts/getedid [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+    fi
+  done
+
+  #create edid
+  mkdir -p /storage/.config/firmware/edid
+  cat "/sys/class/drm/$hdmi/edid" > /storage/.config/firmware/edid/edid.bin
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid]: Binary data... dunno, maybe use `cp` instead?

File: packages/sysutils/busybox/scripts/getedid (hunk 9 above)
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid]: Also, you need to keep the `card0-` prefix, at least on my Skylake, in which case you just need:
```
      hdmi="$(basename "$i")"
```

File: packages/sysutils/busybox/scripts/getedid [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+    intel 
+  elif [ "$gpu" = "nvidia" ]; then
+    nvidia
+  fi
+}
+
+
+# mounting /flash to rw
+mount_rw() {
+  mount -o remout,rw /flash
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid]: `remount`

File: packages/sysutils/busybox/scripts/getedid [hunk 12]
```diff
@@ -0,0 +1,274 @@
...
+  #create cpio archive
+  cd /storage
+  mkdir -p cpio/lib/firmware/edid
+  cp /storage/.config/firmware/edid/edid.bin /storage/cpio/lib/firmware/edid/
+  cd cpio/
+  find . -print | cpio -ov -H newc > /storage/edid.cpio
+
+  #remount /flash to rw
+  mount_rw
+  mv /storage/edid.cpio "$sys_path"
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid]: `sys_path` isn't known at this point, you haven't called `check_file` yet...

File: packages/sysutils/busybox/scripts/getedid [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+  reboot
+}
+
+
+# check syslinux.cfg and/or extlinux.conf
+check_file() {
+  # check boot system
+  if [ -d /sys/firmware/efi]; then
+    boot="UEFI"
+    sys_path="flash/EFI/BOOT"
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid]: Missing `/` - should be:
```
    sys_path="/flash/EFI/BOOT"
```

**Comment by MilhouseVH** [comment]
For intel, are both `/storage/.config/firmware/edid/edid.bin` and `/storage/cpio/lib/firmware/edid/edid.bin` required? Or is the cpio folder just being used temporarily to create the cpio file (in which case, it's probably not required at all)?

**Comment by DaVukovic** [comment]
changes done ^^

**Pull request merged**